### PR TITLE
Add Scotto40 Keyboard

### PR DESCRIPTION
**Pull request by joe-scotto**
Add Scotto40 keyboard and default keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/jscotto/scotto40/info.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        // 2, 3, 4, 5, 6, 7, 8, 9, A3, A2
```
Comment (drashna) [inline keyboards/handwired/jscotto/scotto40/info.json:16]:
```suggestion
```

File: keyboards/handwired/jscotto/scotto40/info.json [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        "cols": ["D1", "D0", "D4", "C6", "D7", "E6", "B4", "B5", "F4", "F5"],
+
+        // 15, 14, 16, 10
```
Comment (drashna) [inline keyboards/handwired/jscotto/scotto40/info.json:19]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/jscotto/scotto40/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+// Tap Dance declarations
```
Comment (drashna) [inline keyboards/handwired/jscotto/scotto40/keymaps/default/keymap.c:20]: Could the tapdance stuff be moved to another keymap? 

We want to keep the default keymap fairly simple, as it's cloned when making a new keymap.  For newer users, complicated keymaps tend to cause more issues than they solve. 

That said, adding a "stock" type keymap with more features is fine.

File: keyboards/handwired/jscotto/scotto40/readme.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Scotto40
+
+![Scotto40](https://i.imgur.com/wtW5xOt.jpeg)
```
Comment (drashna) [inline keyboards/handwired/jscotto/scotto40/readme.md:3]:
```suggestion
![Scotto40](https://i.imgur.com/wtW5xOth.jpeg)
```

File: keyboards/handwired/jscotto/scotto40/keymaps/default/rules.mk [hunk 5]
```diff
@@ -0,0 +1,2 @@
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/handwired/jscotto/scotto40/keymaps/default/rules.mk:1]: this should probably be moved to the keyboard's rules file

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/jscotto/scotto40/keymaps/default/keymap.c (hunk 3 above)
Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto40/keymaps/default/keymap.c:20]: The TD on default is specifically for CMD+SPC and ESC which have nowhere to go on the board, that's why it's there.

File: keyboards/handwired/jscotto/scotto40/keymaps/default/rules.mk (hunk 5 above)
Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto40/keymaps/default/rules.mk]: Good point... done.

**Comment by joe-scotto** [comment]
@drashna This one as well please :)

File: keyboards/handwired/jscotto/scotto40/info.json [hunk 6]
```diff
@@ -0,0 +1,180 @@
+{
```
Comment (noroadsleft) [inline keyboards/handwired/jscotto/scotto40/info.json:1]: Please remove all the inline comments (`// Row 1`, etc.) from this file.

File: keyboards/handwired/jscotto/scotto40/info.json [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+                { "matrix": [2, 7], "x": 7, "y": 2 },
+                { "matrix": [2, 8], "x": 8, "y": 2 },
+                { "matrix": [2, 9], "x": 9, "y": 2 },
+
+                // Row 4
+                { "matrix": [3, 0], "x": 0, "y": 3 },
+
+                { "matrix": [3, 2], "x": 2, "y": 3 },
+                { "matrix": [3, 3], "x": 3, "y": 3 },
+                { "matrix": [3, 4], "x": 4, "y": 3 },
```
Comment (noroadsleft) [inline keyboards/handwired/jscotto/scotto40/info.json:70]:
```suggestion
                { "matrix": [3, 4], "x": 4, "y": 3, "w": 2 },
```

File: keyboards/handwired/jscotto/scotto40/readme.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Scotto40
+
+![Scotto40](https://i.imgur.com/wtW5xOth.jpeg)
+
+A 37, 38, or 40 key handwired ortholinear keyboard. Case files available [here](https://github.com/joe-scotto/keyboards.git).
```
Comment (noroadsleft) [inline keyboards/handwired/jscotto/scotto40/readme.md:5]:
```suggestion
A 37, 38, or 40 key handwired ortholinear keyboard. Case files available [here](https://github.com/joe-scotto/keyboards).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/handwired/jscotto/scotto40/keymaps/default/keymap.c (hunk 3 above)
Comment (noroadsleft) [reply keyboards/handwired/jscotto/scotto40/keymaps/default/keymap.c:20]: Setting the default keymap to use the `LAYOUT_ortho_4x10` macro instead would give you a few extra keys.

Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto40/keymaps/default/keymap.c:20]: The intended mapping is the one I have set, that is just an alternative option the board offers.

File: keyboards/handwired/jscotto/scotto40/info.json (hunk 6 above)
Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto40/info.json:1]: I think for sake of clarity with this being a handwired board, the comments are useful. When wiring it up, it's nice to be able to quickly see where the rows are and what matrix coordinates are needed.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**